src/auth.rs

```diff
@@ -127,25 +129,54 @@ where
     }
 }
 
+trait RoleBounds<Role>: DynClone + Send + Sync {
+    fn contains(&self, role: &Role) -> bool;
+    fn bounds(
+        &self,
+    ) -> (
+        std::collections::Bound<&Role>,
+        std::collections::Bound<&Role>,
+    );
+}
+
+impl<T, Role> RoleBounds<Role> for T
+where
+    Role: PartialOrd + PartialEq + Clone + Send + Sync + 'static,
+    T: RangeBounds<Role> + Clone + Send + Sync,
+{
+    fn contains(&self, role: &Role) -> bool {
+        RangeBounds::contains(self, role)
```

Comment:
```suggestion
    fn contains(&self, role: Option<Role>) -> bool {
        if let Some(role) = role {
            RangeBounds::contains(self, &role)
        } else {
            false
        }
```

Reply:
Thanks! Changed the code according to your suggestions.